src/traffic_server/InkIOCoreAPI.cc

```diff
@@ -634,11 +634,12 @@ TSIOBufferWrite(TSIOBuffer bufp, const void *buf, int64_t length)
 }
 
 // not in SDK3.0
-void
-TSIOBufferReaderCopy(TSIOBufferReader readerp, const void *buf, int64_t length)
+int64_t
+TSIOBufferReaderCopy(TSIOBufferReader readerp, void *buff, int64_t length)
 {
-  IOBufferReader *r = (IOBufferReader *)readerp;
-  r->memcpy(buf, length);
+  auto r{reinterpret_cast<IOBufferReader *>(readerp)};
+  auto limit = r->memcpy(buff, length, 0);
+  return static_cast<char *>(limit) - static_cast<char *>(buff);
```

Comment:
limit is a char*, you shouldn't have to cast It.   We shouldn't use auto for return values from function calls.

Reply:
In my experience `auto` for function call return values is probably the most common usage, since so many functions now return rather complex types (e.g, `begin` and `end` methods). For something like line 640, clang will actually complain about "redundant type definitions".

Reply:
For iterators auto is OK, but for some random method or function we shouldn't use it.

This was a classic example of why not to use auto for a return value from a method, since you were casting it the same type.